controllers/nodehealthcheck_controller.go

```diff
@@ -120,10 +122,16 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		// stop reconciling
 		return result, nil
 	}
-	if meta.IsStatusConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled) {
-		log.Info("re-enabling NHC, no conflicting MHC configured in the cluster")
-		meta.RemoveStatusCondition(&nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled)
-		r.Recorder.Eventf(nhc, eventTypeNormal, eventReasonEnabled, "Custom MachineHealthCheck(s) removed, re-enabling NHC")
+	// update status if needed
+	if !meta.IsStatusConditionFalse(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled) {
```

Comment:
Nit: IIUC 
`!meta.IsStatusConditionFalse(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled) ` 
is equivelent to
 `meta.IsStatusConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled)`
 I'd rather use the later

Reply:
I thought so as well, but `!IsStatusConditionTrue` is also true when the condition does not exist. But I want to see that it exists and is false 🙂